### DISCUSSION TITLE
How to work with two devises and one repository

### DISCUSSION THREAD

**Discussion by pierrot10** [community/community#23754] [New to GitHub]
<p>Hello,</p>
<p>I am novice with Github. So I am using it since a moment, but it first time, I upload my code to my own repository that I created.</p>
<p>On a Rasberry I created a folder</p>
<pre><code>mkdir /opt/project/
</code></pre>
<p>I have created a pythion script with a class.</p>
<p>First time I use it, I was helped with this</p>
<pre><code>sudo git init
sudo git add script-py3.py
sudo git commit -m "first commit"
sudo git config --global user.email "email@email"
sudo git config --global user.name "username"
sudo git commit -m "first commit"
sudo git remote add origin https://github.com/username/projet-v3.git
sudo git push -u origin master
</code></pre>
<p>When I have done with a lot of code or when I wanted to upload new changes, I have used to do it with</p>
<pre><code>sudo git add script-py3.py
sudo git commit -m "update script-py3.py"
sudo git push -u origin master
</code></pre>
<p>and my changes was uploaded at github.</p>
<p>My worries now, I have a new Raspberry and I would like to download the same script. To do it, it’s easy, but I am going to add new line of code and then pusing the change to the same repository, whith the command I know.</p>
<p>Later, I would like to update the script on my first Raspberry, but what will happen in my repository if I push again from my first Raspberry? Do the repository will be overwritten by my old script from my first Pi?</p>
<p>The idea is both Raspberry will download the lastest script version from the repository.</p>
<p>Then from my second Raspberry, after downloading the repository, shoud I</p>
<pre><code>sudo git init
sudo git add script-py3.py
sudo git commit -m "first commit"
sudo git config --global user.email "email@email"
sudo git config --global user.name "username"
sudo git remote add origin https://github.com/username/projet-v3.git
sudo git push -u origin master
</code></pre>
<p>and then after new changes</p>
<pre><code>sudo git add script-py3.py
sudo git commit -m "update script-py3.py"
sudo git push -u origin master
</code></pre>
<p>But then from my first, how can I update the change from the depository to the Raspberry?</p>
<p>And if I add changes from the first Pi and I push, how to update the second without overwrite the latest version of the repository from a local version?</p>
<p>I am starting to install my second Raspberry but I am afraid of a wrong git manipulation :slight_smile:</p>
<p>Many thanks</p>

**Comment by yamiacat** [comment]
<p>Hi pierrot10! It looks like you’ve got the hang of making changes to your code, committing it, and pushing the code up to the GitHub remote!</p>
<p>The missing piece of the puzzle is <code>git pull</code>!</p>
<p>Each time you start working with either one of the Raspberry Pis, you can run <code>git pull</code> to update your local copy of the repository with any changes the GitHub version might have had pushed to it since the last time you worked on that Pi.</p>
<p>This help page has more details:</p>
<p><a href="https://help.github.com/en/github/using-git/getting-changes-from-a-remote-repository" class="onebox" target="_blank" rel="nofollow noopener">https://help.github.com/en/github/using-git/getting-changes-from-a-remote-repository</a></p>

**Comment by pierrot10** [comment]
<p>Hello<br>
Thanks for your reply. So do you mean, I should do that?</p>
<pre><code class="lang-auto">sudo git pull -u origin master
</code></pre>
<p>That’s correct? But if I want to pull just on file, I supposed I can do as the following?</p>
<p>sudo git pull <a href="https://github.com/user/repo/README.md" rel="nofollow noopener">https://github.com/user/repo/README.md</a></p>

**Comment by yamiacat** [comment, marked as answer]
<p>Hi there!</p>
<p>The <code>-u</code> flag is short for <code>--upstream</code> - it tells Git “remember this relationship for future use”, so when you have used it once (usually on your first push) you don’t need to use it again.</p>
<p><code>git pull origin master</code> is the most common workflow, and it will get all changes to the master branch and merge them in to your local copy. Mostly people prefer to get all changes from origin at the same time.</p>
<p><code>git pull</code> is actually a combination of two commands, <code>fetch</code> and <code>merge</code>, that you can run separately.</p>
<p><code>git fetch origin master</code> will find out about all available changes to the master branch.</p>
<p><code>git checkout FETCH_HEAD -- &lt;specific file&gt;</code>will grab only the specified file.</p>
<p>You’ll then need to commit that file separately.</p>
<p>As I mentioned though, it’s much more usual to just grab all changes, so <code>git pull</code> does all of that in one  command.</p>
